File: credit-engine-api/src/main/java/com/credit/engine/application/dto/currency/CurrencyRequest.java
package com.credit.engine.application.dto.currency;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record CurrencyRequest(
        @NotBlank(message = "O código da moeda é obrigatório.")
        @Size(min = 3, max = 3, message = "O código da moeda deve possuir exatamente 3 letras (ex: BRL, USD).")
        String code,

        @NotBlank(message = "O nome da moeda é obrigatório.")
        @Size(max = 50, message = "O nome da moeda deve conter no máximo 50 caracteres.")
        String name,

        @NotBlank(message = "O símbolo da moeda é obrigatório.")
        @Size(max = 5, message = "O símbolo da moeda deve conter no máximo 5 caracteres.")
        String symbol
) {
}

File: credit-engine-ui/src/services/currency/dto/CurrencyRequest.ts
import ValidationUtils from '../../../utils/ValidationUtils.ts';

export default interface CurrencyRequest {
    code: string;
    name: string;
    symbol: string;
}

export const validateCurrencyRequest = (request: CurrencyRequest): { message: string }[] => {
    const errors: { message: string }[] = [];

    if (ValidationUtils.isBlank(request.code) || request.code.trim().length !== 3)
        errors.push({ message: 'O código da moeda deve possuir exatamente 3 letras (ex: BRL, USD).' });

    if (ValidationUtils.isBlank(request.name))
        errors.push({ message: 'O nome da moeda e obrigatório.' });

    if (ValidationUtils.isBlank(request.symbol))
        errors.push({ message: 'O simbolo da moeda e obrigatório.' });

    return errors;
};
